### ISSUE TITLE
Support multiple dice types in a roll request (Earthdawn dice)

### ISSUE THREAD

**Issue by Elfabet** [ArtemGr/Sidekick#53]
Can we get support for /r D6+D4, and a sum total.
Shadows of the Demon Lord uses multiple dice types as well in a line.

Thanks,
-Jake

**Comment by ArtemGr** [comment]
Could you give me an example?

**Comment by Elfabet** [comment]
yeah, so right now if you ask it to roll two different sizes of dice in the same line (e.g. a D4 and a D6), it gives me one of its clever 'does not comupte' throwbacks.

So for example:
/r D20+5+D6
would be awesome if it returned:
@user: D20+5+D6 = (17)+5+(5) = 27

even more ideally, you'd include all the other regular expressions like Keep, and !explode, calculated for each dice set as well.
/r 4d20!+3d6K1+6
@user: (2+2+4+20+1)+(4+1+4+1)+6=39 
(sorry, don't know strike through text)

**Comment by ArtemGr** [comment]
I see, thanks!

So the immediate problem is using the upper-case letters. For example, the following lowercased specification works: "/r 4d20!+3d6k1+6" (it gives me (10+10+17+12)+(~~2~~+3+~~2~~)+6 = 58)).

Can you comment on the upper-case letters? Do you find them more readable? Are they customary in a role-playing system you use?

**Comment by Elfabet** [comment]
Oh snap! Wha~~

No problem then!
I consider capital and non-capital d's with numbers on either side interchangable, but I see how that could be problematic. 
Um, way back when I was writing my own system in excel, it would autocapitalize my dice codes, so I just got in the habit, I guess.
This works beautifully and wonderful, and this 'issue' is more than already addressed.
Thanks!

**Comment by ArtemGr** [comment]
You're welcome!

> Um, way back when I was writing my own system in excel, it would autocapitalize my dice codes, so I just got in the habit, I guess.

My problem is that I hope to add more systems to the bot, systems I don't yet even know about.
The [dice notation](https://en.wikipedia.org/wiki/Dice_notation) tries to make a shorthand for everything. One can easily imagine a situation where there is no more small letter characters to pick from and new functions are being added on the capital ones.
Or a role-playing system which has a syntax that is case-sensitive.

I'll think about automatic lowercasing.
/r D6, at least, should work just as well as /r d6.
